astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,385 @@
+            filename : string, file name to be used to store the postcard,
+            optional, default None
+            File name for the artifact
+            verbose : bool, optional, default 'False'
+            Flag to display information about the process
+
+            Returns
+            -------
+            None. It downloads the observation postcard indicated
+        """
+
+        retri_type = "RETRIEVAL_TYPE=POSTCARD"
+        obs_id = "OBSERVATION_ID=" + observation_id
+        cal_level = "CALIBRATION_LEVEL=" + calibration_level
+        res = "RESOLUTION=" + str(resolution)
+        link = "".join((self.data_url,
+                        retri_type,
+                        "&",
+                        obs_id,
+                        "&",
+                        cal_level,
+                        "&",
+                        res))
+        if verbose:
+            log.info(link)
+        if filename is None:
+            filename = observation_id + ".tar"
+        self._handler.get_file(link, filename, verbose)
+
+    def cone_search(self, coordinates, radius=0.0, filename=None,
+                    output_format='votable', verbose=False):
+        coord = self._getCoordInput(coordinates, "coordinate")
+        radiusInGrades = float(radius/60)  # Converts to degrees
+
+        raHours, dec = commons.coord_to_radec(coord)
+        ra = raHours * 15.0  # Converts to degrees
+        payload = {"RESOURCE_CLASS": "OBSERVATION",
+                   "ADQLQUERY": "SELECT DISTINCT OBSERVATION,OBSERVATION.TYPE,"
+                   "TARGET.MOVING_TARGET"
+                   ",TARGET.TARGET_NAME,TARGET.TARGET_DESCRIPTION,PROPOSAL."
+                   "PROPOSAL_ID,PROPOSAL.PI_"
+                   "NAME,PROPOSAL.PROPOSAL_TITLE,INSTRUMENT.INSTRUMENT_NAME,"
+                   "PLANE.METADATA_PROVENANCE"
+                   ",PLANE.DATA_PRODUCT_TYPE,PLANE.SOFTWARE_VERSION,POSITION"
+                   ".RA,POSITION.DEC,POSITION."
+                   "GAL_LAT,POSITION.GAL_LON,POSITION.ECL_LAT,POSITION.ECL_LON"
+                   ",POSITION.FOV_SIZE,ENERGY."
+                   "WAVE_CENTRAL,ENERGY.WAVE_BANDWIDTH,ENERGY.WAVE_MAX,ENERGY"
+                   ".WAVE_MIN,ENERGY.FILTER FROM"
+                   " FIELD_NOT_USED  WHERE OBSERVATION.COLLECTION='HST'  AND  "
+                   "PLANE.MAIN_SCIENCE_PLANE="
+                   "'true'  AND  (OBSERVATION.TYPE='HST Composite' OR "
+                   "OBSERVATION.TYPE='HST Singleton')"
+                   "  AND  INTERSECTS(CIRCLE('ICRS'," + str(ra) + ""
+                   "," + str(dec) + "," + str(radiusInGrades) + ""
+                   "),POSITION)=1  AND  PLANE.MAIN_SCIENCE_PLANE='true' "
+                   "ORDER BY PROPOSAL.PROPOSAL_ID "
+                   "DESC",
+                   # "PAGE": "1",
+                   # "PAGE_SIZE": "50",
+                   "RETURN_TYPE": str(output_format)}
+        result = urllib.parse.urlencode(payload)
```

Comment:
If this is the instance of `urlencode` you were talking about, it simply isn't needed.  You can do something like:
```
request.get(base_url, data=payload)
```
which send the `payload` as part of the URL, e.g.
```
payload = {'key':'val'}
response = requests.get('http://httpbin.org/get', params=payload)
print(response.url)
```
the print will be `http://httpbin.org/get?key=val`.

In astroquery, instead of `requests.get`, do:
`self._request('GET', url, params=...)`

Reply:
The approach I recommend will perform the request; it's not just assembling the URL.

Reply:
Yes, I know. I will change that tomorrow. I just wanted to make sure the rest is fine